lib/netext/dialer.go

```diff
@@ -32,27 +33,42 @@ import (
 	"github.com/loadimpact/k6/lib/metrics"
 	"github.com/loadimpact/k6/stats"
 
-	"github.com/viki-org/dnscache"
+	"github.com/domainr/dnsr"
+	"github.com/miekg/dns"
 )
 
+var ip6 bool
+var ip4 bool
+
 // Dialer wraps net.Dialer and provides k6 specific functionality -
 // tracing, blacklists and DNS cache and aliases.
 type Dialer struct {
 	net.Dialer
 
-	Resolver  *dnscache.Resolver
+	Resolver  *dnsr.Resolver
+	IP4       map[string]bool // IPv4 last seen
+	CNAME     map[string]CanonicalName
 	Blacklist []*lib.IPNet
 	Hosts     map[string]net.IP
 
 	BytesRead    int64
 	BytesWritten int64
 }
 
+// CanonicalName is an expiring CNAME value.
+type CanonicalName struct {
+	Name   string
+	TTL    time.Duration
+	Expiry time.Time
+}
+
 // NewDialer constructs a new Dialer and initializes its cache.
 func NewDialer(dialer net.Dialer) *Dialer {
 	return &Dialer{
 		Dialer:   dialer,
-		Resolver: dnscache.New(0),
+		Resolver: dnsr.NewExpiring(0),
```

Comment:
here `0` means `1000` given the [current code](https://github.com/domainr/dnsr/blob/44884fe6c3890c5941e57f260a6ac67f5c3ecc56/cache.go#L17-L30) of dnsr. Can you either add a comment for this or actually put a 1000?